src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -147,7 +147,69 @@ public RestAzureNS.AzureOperationResponse TriggerBackup()
 
         public RestAzureNS.AzureOperationResponse TriggerRestore()
         {
-            throw new NotImplementedException();
+            string vaultName = (string)ProviderData[VaultParams.VaultName];
+            string resourceGroupName = (string)ProviderData[VaultParams.ResourceGroupName];
+            CmdletModel.AzureFileShareRecoveryPoint recoveryPoint = ProviderData[RestoreBackupItemParams.RecoveryPoint]
+                as CmdletModel.AzureFileShareRecoveryPoint;
+            string storageAccountId = ProviderData[RestoreBackupItemParams.StorageAccountId].ToString();
+            string storageAccountLocation = ProviderData.ContainsKey(RestoreBackupItemParams.StorageAccountLocation) ?
+                ProviderData[RestoreBackupItemParams.StorageAccountLocation].ToString() : null;
+            string copyOptions = (string)ProviderData[RestoreFSBackupItemParams.ResolveConflict];
+            string sourceFilePath = ProviderData.ContainsKey(RestoreFSBackupItemParams.SourceFilePath) ?
+                (string)ProviderData[RestoreFSBackupItemParams.SourceFilePath] : null;
+            string sourceFileType = ProviderData.ContainsKey(RestoreFSBackupItemParams.SourceFileType) ?
+                (string)ProviderData[RestoreFSBackupItemParams.SourceFileType] : null;
+            string targetStorageAccountName =
+                ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetStorageAccountName) ?
+                (string)ProviderData[RestoreFSBackupItemParams.TargetStorageAccountName] : null;
+            string targetFileShareName = ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetFileShareName) ?
+                (string)ProviderData[RestoreFSBackupItemParams.TargetFileShareName] : null;
+            string targetFolder = ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetFolder) ?
+                (string)ProviderData[RestoreFSBackupItemParams.TargetFolder] : null;
+            string targetStorageAccountId = ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetStorageAccountId) ?
+                (string)ProviderData[RestoreFSBackupItemParams.TargetStorageAccountId] : null;
+            string targetStorageAccountLocation = ProviderData.ContainsKey(RestoreFSBackupItemParams.TargetStorageAccountLocation) ?
+                ProviderData[RestoreFSBackupItemParams.TargetStorageAccountLocation].ToString() : null;
+
+            List<RestoreFileSpecs> restoreFileSpecs = null;
+            if (sourceFilePath != null || targetFolder != null)
```

Comment:
Shouldn't this be an "&&" check? Also, if both are null, restoreFileSpecs would be null. What will happen when we pass this in the Restore API? Should we throw an error instead?

Reply:
It will be an OR check because it is possible to have a target folder but no sourceFilePath(in the case of alternate location full restore)
restoreFileSpecs needs to be kept null in the case of original location full restore

Reply:
That's efficient but not much readable. Can we have 4 separate if-else conditions one for each Restore scenario and explicitly set the correct parameters? This is just for better readability.